components/style/values.rs

```diff
@@ -1477,6 +1477,43 @@ pub mod specified {
     }
 
     #[derive(Clone, Copy, Debug, PartialEq, PartialOrd, HeapSizeOf)]
+    pub struct Number(pub CSSFloat);
+
+    impl Number {
+        pub fn parse(input: &mut Parser) -> Result<Number, ()> {
+            parse_number(input).map(Number)
+        }
+
+        fn parse_with_minimum(input: &mut Parser, min: CSSFloat) -> Result<Number, ()> {
+            match parse_number(input) {
+                Ok(value) if value < min => Err(()),
+                value => value.map(Number),
```

Comment:
There are a couple of other similar uses in this file – mapping to Au and Percentage – and a bunch more mapping to enum constructors, which is not so different.